sound/soc/sof/intel/cnl.c

```diff
@@ -164,9 +167,29 @@ static int cnl_ipc_send_msg(struct snd_sof_dev *sdev,
 
 static void cnl_ipc_dump(struct snd_sof_dev *sdev)
 {
+	struct hdac_bus *bus = sof_to_bus(sdev);
 	u32 hipcctl;
 	u32 hipcida;
 	u32 hipctdr;
+	u32 adspis;
+	u32 intsts;
+	u32 intctl;
+	u32 ppsts;
+	u8 rirbsts;
+
+	/* read key IRQ stats and config registers */
+	adspis = snd_sof_dsp_read(sdev, HDA_DSP_BAR, HDA_DSP_REG_ADSPIS);
+	intsts = snd_sof_dsp_read(sdev, HDA_DSP_HDA_BAR, SOF_HDA_INTSTS);
+	intctl = snd_sof_dsp_read(sdev, HDA_DSP_HDA_BAR, SOF_HDA_INTCTL);
+	ppsts = snd_sof_dsp_read(sdev, HDA_DSP_PP_BAR, SOF_HDA_REG_PP_PPSTS);
+	rirbsts = snd_hdac_chip_readb(bus, RIRBSTS);
+
+	dev_err(sdev->dev,
+		"error: hda irq intsts 0x%8.8x intlctl 0x%8.8x rirb %2.2x\n",
+		intsts, intctl, rirbsts);
+	dev_err(sdev->dev,
+		"error: dsp irq ppsts 0x%8.8x adspis 0x%8.8x\n",
+		ppsts, adspis);
```

Comment:
useful info but it would be nice to move it to a helper that can be called from the CNL or APL IPC? This isn't CNL-specific at all.

Reply:
@plbossart Ack, moved to common code and tested on APL (UP2).